array_schema_evolution_test.go

```diff
@@ -0,0 +1,147 @@
+//go:build experimental
+// +build experimental
+
+package tiledb
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const array_schema_evolution_name = "array_schema_evolution"
+
+func TestArraySchemaEvolution(t *testing.T) {
+	config, err := NewConfig()
+	assert.Nil(t, err)
+	assert.NotNil(t, config)
+
+	// Test context with config
+	context, err := NewContext(config)
+	assert.Nil(t, err)
+	assert.NotNil(t, context)
+
+	dimension, err := NewDimension(context, "test", TILEDB_INT32, []int32{1, 10}, int32(5))
+	assert.Nil(t, err)
+	assert.NotNil(t, dimension)
+
+	domain, err := NewDomain(context)
+	assert.Nil(t, err)
+	assert.NotNil(t, domain)
+
+	// Add dimension to domain
+	err = domain.AddDimensions(dimension)
+	assert.Nil(t, err)
+
+	arraySchema, err := NewArraySchema(context, TILEDB_DENSE)
+	assert.Nil(t, err)
+	assert.NotNil(t, arraySchema)
+
+	a1, err := NewAttribute(context, "a1", TILEDB_INT32)
+	assert.Nil(t, err)
+	assert.NotNil(t, a1)
+
+	a2, err := NewAttribute(context, "a2", TILEDB_INT32)
+	assert.Nil(t, err)
+	assert.NotNil(t, a2)
+
+	err = arraySchema.AddAttributes(a1, a2)
+	assert.Nil(t, err)
+
+	err = arraySchema.SetCapacity(100)
+	assert.Nil(t, err)
+
+	err = arraySchema.SetDomain(domain)
+	assert.Nil(t, err)
+
+	err = arraySchema.Check()
+	assert.Nil(t, err)
+
+	array, err := NewArray(context, array_schema_evolution_name)
+	assert.Nil(t, err)
+	assert.NotNil(t, array)
+
+	defer os.RemoveAll(array_schema_evolution_name)
+
+	err = array.Create(arraySchema)
+	assert.Nil(t, err)
+
+	err = array.Close()
+	assert.Nil(t, err)
+
+	arraySchemaEvolution, err := NewArraySchemaEvolution(context)
+	assert.Nil(t, err)
+	defer arraySchemaEvolution.Free()
```

Comment:
Since the finalizer is automatically called by the GC, this should not be needed.